lib/orogen/spec/task_context.rb

```diff
@@ -606,20 +606,14 @@ def needs_configuration
             #   attribute('device_name', '/std/string/, '').
             #       doc 'the device name to connect to'
             def attribute(name, type, default_value = nil)
-                @attributes[name] = att = configuration_object(Attribute, name, type, default_value)
+                att = configuration_object(Attribute, name, type, default_value)
+                @attributes[name] = att
                 Spec.load_documentation(att, /attribute/)
                 att
             end
 
             def configuration_object(klass, name, type, default_value)
-                name = OroGen.verify_valid_identifier(name)
-                check_uniqueness(name)
-
-                begin
-                    type = project.find_interface_type(type)
-                rescue Typelib::NotFound => e
-                    raise ConfigError, "invalid type #{type}: #{e.message}", e.backtrace
-                end
+                type = resolve_type_if_needed(type)
```

Comment:
what do you think about `#resolve_type_if_name`?

Reply:
Actually, I got rid of it altogether and call project.resolve_type in all cases.